### ISSUE TITLE
Question about GAT and GCN layers

### ISSUE THREAD

**Issue by goolhanrry** [rusty1s/pytorch_geometric#867]
## ❓ Questions & Help

Hi @rusty1s , Thanks for your great work! I'm using PyG in my project and I have a question about GAT.

In your example (as well as in the GAT paper), an extra dropout layer has been used before each GAT layer, to cope with the small training set in Cora dataset. My question is, in my own project, I'd like to use 3-layer GAT to map the node features, and all the nodes are used for training. In this case, since the training set is big enough, should I still used the two dropouts (the one before GAT layer and the one after computing attention coefficients) or I can try simply remove them?

I've tried to keep the dropouts but the performance is even worse than the normal GCNConv layers. I really appreciate it if you can give me some advice : )

**Comment by rusty1s** [comment]
As you said, the dropout layers are mainly used to cope with the small training set. You can, of course, simply remove them if you feel that your model is underperforming.

**Comment by goolhanrry** [comment]
Thank you so much for your early reply! I am concerning another problem with GCN.

As described in papers, GCN can not handle directed graphs because it doesn't make sense (even if the implementation works), and it seems that edge weights are not used for GCN as well. However, GCN with directed graphs in my project did work better than the one with undirected graphs, and I am concerning how to explain the results.

I haven't found any paper using GCNs on directed graphs or graphs with edge weights. Could you please give me some keywords about the papers using GCN in the cases above if you have met before? Thank you!

**Comment by rusty1s** [comment]
I do not have any specific pointers for you, but it is reasonable to expect that GCN on directed edges works better if this information is crucial to the model's performance. Note that edge weights are used in GCN, just not in the conducted experiments. It works by computing the degree matrix based on the edge weights.

**Comment by goolhanrry** [comment]
Thank you so much!
